packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -166,17 +167,34 @@ export default function useBlockDropZone( {
 		[ targetRootClientId ]
 	);
 
-	const { getBlockListSettings, getBlocks, getBlockIndex } =
-		useSelect( blockEditorStore );
+	const {
+		getBlockListSettings,
+		getBlocks,
+		getBlockIndex,
+		getDraggedBlocksOrigin,
+	} = useSelect( blockEditorStore );
+
 	const { showInsertionPoint, hideInsertionPoint } =
 		useDispatch( blockEditorStore );
 
 	const onBlockDrop = useOnBlockDrop( targetRootClientId, dropTarget.index, {
 		operation: dropTarget.operation,
+		dropZoneName,
 	} );
 	const throttled = useThrottle(
 		useCallback(
 			( event, ownerDocument ) => {
+				const draggedBlocksOrigin = getDraggedBlocksOrigin();
+
+				if (
+					dropZoneName &&
+					draggedBlocksOrigin &&
+					dropZoneName !== draggedBlocksOrigin
+				) {
+					// If drag and drop names are available and they don't match, don't allow dropping.
+					return;
```

Comment:
Exiting early here means that no _drop_ zone UI affordances are displayed when dragging. For example the blue lines that appear when dragging over a drop zone will not show.